lib/components/Input/search-field/__tests__/search-field.test.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ComponentProps } from "react";
+import { render, fireEvent, screen, act } from "@testing-library/react";
+import SearchField from "..";
+
+jest.mock("@deriv/quill-icons", () => ({
+    ...jest.requireActual("@deriv/quill-icons"),
+    StandaloneCircleCheckBoldIcon: jest.fn(() => "mockedSuccessIcon"),
+    StandaloneSearchRegularIcon: jest.fn(() => "mockedSearchIcon"),
+    StandaloneTriangleExclamationBoldIcon: jest.fn(() => "mockedErrorIcon"),
+    StandaloneCircleXmarkFillIcon: jest.fn(() => "mockedClearIcon"),
+}));
+
+describe("<SearchField />", () => {
+    const renderSearchField = (
+        props: ComponentProps<typeof SearchField> = {},
+    ) => {
+        return render(<SearchField {...props} />);
+    };
+    it("renders correctly with default props", () => {
+        const { container } = renderSearchField();
+        expect(screen.getByText("mockedSearchIcon")).toBeInTheDocument();
+        expect(container).toMatchSnapshot();
+    });
+    it("updates value correctly on input change", () => {
+        const onChangeMock = jest.fn();
+        const { container } = renderSearchField({ onChange: onChangeMock });
+        const searchElement = screen.getByPlaceholderText("Search");
+
+        act(() => {
+            fireEvent.change(searchElement, { target: { value: "test" } });
+        });
+
+        expect(onChangeMock).toHaveBeenCalledTimes(1);
+        expect(onChangeMock).toHaveBeenCalledWith(
+            expect.objectContaining({
+                target: expect.objectContaining({
+                    value: "test",
+                }),
+            }),
+        );
+        expect(container).toMatchSnapshot();
+    });
+    it("renders clear icon when value is not empty and clear icon should be removed after click", () => {
+        const { container } = renderSearchField({ value: "test" });
+        const mockedClearIcon = screen.getByText("mockedClearIcon");
+        expect(mockedClearIcon).toBeInTheDocument();
+
+        act(() => fireEvent.click(mockedClearIcon));
```

Comment:
```suggestion
        await userEvent.click(mockedClearIcon);
```
`userEvent` is considered to be a better approach since it dispatches events like they would happen if a user interacted with the component. is it possible to use `userEvent` here and at L 30? :)